### PR TITLE
ear2ctl: init at 0.1.0

### PR DESCRIPTION
**Pull request by jaredmontoya**
## Description of changes
Added ear2ctl package, a cli tool for controlling Nothing ear earbuds over bluetooth.

I used `nix-init` to generate this package and checked it with `nixpkgs-fmt`.
Then built with `nix run nixpkgs#nixpkgs-review rev HEAD` on my x86_64-linux machine and ran the resulting executable with the --help flag.

The maintainer list is not set because I don't want to block this package's updates made by someone else in case I do not respond to the ping in the future. If this is a requirement, I will fill it out.

This is my first contribution to nixpkgs. If I made any mistakes please kindly point me to them.

## Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- For non-Linux: Is sandboxing enabled in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
  - [ ] `sandbox = relaxed`
  - [ ] `sandbox = true`
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [x] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [24.11 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2411.section.md) (or backporting [23.11](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) and [24.05](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2405.section.md) Release notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

---

Add a :+1: [reaction] to [pull requests you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[pull requests you find important]: https://github.com/NixOS/nixpkgs/pulls?q=is%3Aopen+sort%3Areactions-%2B1-desc

### CODE REVIEW

**Comment by Aleksanaa** [comment]
> If this is a requirement, I will fill it out.

Yes, it's a requirement for at least new packages. See https://github.com/NixOS/nixpkgs/tree/master/pkgs#meta-attributes

File: pkgs/by-name/ea/ear2ctl/package.nix [hunk 1]
```diff
@@ -0,0 +1,27 @@
...
+  };
+
+  cargoHash = "sha256-ax+/lvdEOjLnwE3Gvji7aaeF9KXjoOXdlTvxYDo8wGI=";
+
+  nativeBuildInputs = [ pkg-config ];
+
+  buildInputs = [ dbus ];
+
+  meta = with lib; {
+    description = "Controls for the Nothing Ear (2)";
```
Comment (Aleksanaa) [inline pkgs/by-name/ea/ear2ctl/package.nix:21]:
```suggestion
    description = "Linux controller for the Nothing Ear (2)";
```

File: pkgs/by-name/ea/ear2ctl/package.nix [hunk 2]
```diff
@@ -0,0 +1,27 @@
...
+    hash = "sha256-xaxl4opLMw9KEDpmNcgR1fBGUqO4BP5a/U52Kz+GAvc=";
+  };
+
+  cargoHash = "sha256-ax+/lvdEOjLnwE3Gvji7aaeF9KXjoOXdlTvxYDo8wGI=";
+
+  nativeBuildInputs = [ pkg-config ];
+
+  buildInputs = [ dbus ];
+
+  meta = with lib; {
```
Comment (Aleksanaa) [inline pkgs/by-name/ea/ear2ctl/package.nix:20]: Better to avoid `with lib;`. I'm not sure why nix-init still generates that.

**Comment by jaredmontoya** [comment]
@Aleksanaa I think everything should be fine now

**Comment by Aleksanaa** [comment]
Oh I'm fucked up. Remember to squash your commits next time.

**Pull request merged**